### PR TITLE
TMCS stopping, 1 of ...

### LINKED ISSUE #241: Implement stopping criterion from TMC paper
(instead of stddev)

### PR DESCRIPTION
**Pull request by mdbenito**
### Description

This PR is a first step towards implementing general stopping criteria

### Changes

- Adds operations on ValuationResults and (Valuation)Status
- Renames ValuationStatus to Status (since it's independent of the algorithm)
- Moves a couple of classes around to avoid circular imports
- Does some cleanup

### Checklist

- [x] Wrote Unit tests (if necessary)
- [x] Updated Documentation (if necessary)
- [x] Updated Changelog
- [x] If notebooks were added/changed, added boilerplate cells are tagged with `"nbsphinx":"hidden"`

### CODE REVIEW

File: src/pydvl/value/shapley/montecarlo.py [hunk 1]
```diff
@@ -40,23 +41,21 @@
...
-    ParallelConfig,
-    Utility,
-    get_running_avg_variance,
-    init_parallel_backend,
-    maybe_progress,
-    random_powerset,
-)
-from pydvl.value import ValuationResult, ValuationStatus
-
+from ...utils.config import ParallelConfig
```
Comment (AnesBenmerzoug) [inline src/pydvl/value/shapley/montecarlo.py:44]: Seeing this makes me realize that absolute imports are more readable.

File: src/pydvl/value/shapley/types.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from enum import Enum
+from typing import Protocol
+
+import numpy as np
+from numpy._typing import NDArray
+
+
+class ShapleyMode(str, Enum):
```
Comment (AnesBenmerzoug) [inline src/pydvl/value/shapley/types.py:8]: Moving this here makes it a bit more cumbersome to import and use for the users.

**Review by AnesBenmerzoug — APPROVED** [review]
Looks good to me.

I left 2 comments but feel free to ignore them for now.

File: src/pydvl/value/shapley/montecarlo.py (hunk 1 above)
Comment (mdbenito) [reply src/pydvl/value/shapley/montecarlo.py:44]: Agreed, I think pycharm suggested it and I kept using them, thinking "maybe we'd want to rearrange stuff later on", but I'll revert to absolute ones in the 2nd PR

File: src/pydvl/value/shapley/types.py (hunk 2 above)
Comment (mdbenito) [reply src/pydvl/value/shapley/types.py:8]: I think I moved it to avoid some circular import. I'll try to move it back in the next PR.

**Pull request merged**